internal/servers/plugin/v3/plugin.go

```diff
@@ -402,12 +402,11 @@ func (s *Server) Write(stream pb.Plugin_WriteServer) error {
 
 func (s *Server) Transform(stream pb.Plugin_TransformServer) error {
 	var (
-		recvRecords       = make(chan arrow.Record)
-		sendRecords       = make(chan arrow.Record)
-		pluginStopsWriter = make(chan struct{})
-		doneReading       = false
-		ctx               = stream.Context()
-		eg, gctx          = errgroup.WithContext(ctx)
+		recvRecords = make(chan arrow.Record)
+		sendRecords = make(chan arrow.Record)
+		pluginStops = make(chan error)
```

Comment:
Do we want to close `pluginStops` somewhere in this function? Also who's closing `recvRecords`?